packages/main/src/AINoticeIndicator.ts

```diff
@@ -0,0 +1,201 @@
+import UI5Element from "@ui5/webcomponents-base/dist/UI5Element.js";
+import customElement from "@ui5/webcomponents-base/dist/decorators/customElement.js";
+import property from "@ui5/webcomponents-base/dist/decorators/property.js";
+import jsxRenderer from "@ui5/webcomponents-base/dist/renderer/JsxRenderer.js";
+import AINoticeIndicatorTemplate from "./AINoticeIndicatorTemplate.js";
+import AINoticeIndicatorDisplayMode from "./types/AINoticeIndicatorDisplayMode.js";
+import slot from "@ui5/webcomponents-base/dist/decorators/slot.js";
+import Link from "./Link.js";
+import Label from "./Label.js";
+import Popover from "./Popover.js";
+import Icon from "./Icon.js";
+import Button from "./Button.js";
+import ResizeHandler from "@ui5/webcomponents-base/dist/delegate/ResizeHandler.js";
+import type { ResizeObserverCallback } from "@ui5/webcomponents-base/dist/delegate/ResizeHandler.js";
+import event from "@ui5/webcomponents-base/dist/decorators/event-strict.js";
+import type { UI5CustomEvent } from "@ui5/webcomponents-base";
+
+// Styles
+import AINoticeIndicatorCss from "./generated/themes/AINoticeIndicator.css.js";
+
+/**
+ * @class
+ * ### Overview
+ * AI Notice control is identifying AI-generated text, providing explanations, or collecting user feedback.
+ *
+ * <h3>Responsive Behavior</h3>
+ *
+ * The AI notice can be shortened if the available horizontal space becomes insufficient due to resizing of the screen or collision with bordering UI elements.
+ *
+ * ### Usage
+ *
+ * Use for AI-specific interactions like identifying AI-generated text, providing explanations, or collecting user feedback.
+ *
+ *
+ * ### ES6 Module Import
+ *
+ * `import "@ui5/webcomponents/dist/AINoticeIndicator.js";`
+ * @constructor
+ * @extends UI5Element
+ * @since 2.12.0
+ * @public
+ */
+@customElement({
+	tag: "ui5-ai-notice-indicator",
+	renderer: jsxRenderer,
+	template: AINoticeIndicatorTemplate,
+	styles: AINoticeIndicatorCss,
+	dependencies: [
+		Label,
+		Popover,
+		Icon,
+		Button,
+		Link,
+	],
+})
+/**
+ * Fired on mouseup, space and enter if ai-notice is interactive
+ *
+ * **Note:** The event will not be fired if the `disabled`
+ * property is set to `true`.
+ * @public
+ */
+@event("click")
```

Comment:
rename the event to something like noticePress because "click" can interfere with native event.